plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,276 @@
+package openweathermap
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type OpenWeatherMap struct {
+	BaseUrl string
+	AppId   string
+	Cities  []string
+
+	client *http.Client
+
+	ResponseTimeout internal.Duration
+}
+
+var sampleConfig = `
+  ## Root url of weather map REST API
+  base_url = "http://api.openweathermap.org/"
+  # Your personal user token from openweathermap.org
+  app_id = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+  cities = ["2988507", "2988588"]
+
+  # HTTP response timeout (default: 5s)
+  response_timeout = "5s"
+`
+
+func (n *OpenWeatherMap) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *OpenWeatherMap) Description() string {
+	return "Read current weather and forecasts data from openweathermap.org"
+}
+
+func (n *OpenWeatherMap) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	base, err := url.Parse(n.BaseUrl)
+	if err != nil {
+		return err
+	}
+
+	// Create an HTTP client that is re-used for each
+	// collection interval
+
+	if n.client == nil {
+		client, err := n.createHttpClient()
+		if err != nil {
+			return err
+		}
+		n.client = client
+	}
+
+	for _, city := range n.Cities {
```

Comment:
You can batch these calls for city data: https://openweathermap.org/current#severalid

Reply:
OK, good point. I need a bit more time to test this.

Reply:
done in the last commit